src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -75,37 +60,300 @@ class NotInDeployRootError(ClickException):
     (use "./" instead to copy into the deploy root).
     """
 
-    artifact_src: str
-    dest_path: Path
+    dest_path: Union[str, Path]
     deploy_root: Path
 
-    def __init__(self, artifact_src: str, dest_path: Path, deploy_root: Path):
+    def __init__(self, dest_path: Union[Path, str], deploy_root: Path):
         super().__init__(
             f"""
             {self.__doc__}
-            artifact_src = {artifact_src}
             dest_path = {dest_path}
             deploy_root = {deploy_root}
             """
         )
-        self.artifact_src = artifact_src
         self.dest_path = dest_path
         self.deploy_root = deploy_root
 
 
 @dataclass
 class ArtifactMapping:
     """
-    Used to keep track of equivalent paths / globs so we can copy
-    artifacts from the project folder to the deploy root.
+    Used to keep track of equivalent paths / globs so we can copy artifacts from the project folder to the deploy root.
```

Comment:
Aren't we getting rid of this in favor of `PathMapping` pydantic model?

Reply:
In due time, this PR is already big enough. I have a few follow-ups planned.